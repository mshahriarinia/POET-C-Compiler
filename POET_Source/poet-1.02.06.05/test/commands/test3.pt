include Cfront.code

<eval DEBUG{ 
  a = 5;
  if (a > 0) b = a + 1;  else b = a -1;
  "abc" : EXP;
  5 : EXP;
}/>

<eval DEBUG {
  input = (1 2 3 4 7);
  for (inputp = input; inputp != ""; inputp = cdr(inputp)) {
     curelem = car(inputp);
     if (curelem == 1) CONTINUE;
     if (curelem == 3) BREAK;
  }
}
/>

<eval DEBUG{
   input = Nest#(Loop#("i",0,100,1),Nest#(Loop#("j",1,10,1), Stmt#1));
   foreach (input : (curLoop = Loop#_) : TRUE) {
      if (curLoop[Loop.i] == "j") CONTINUE;
   }
   loopNests = "";
   foreach (input : (curNest = Nest#((Loop#_),_)) : FALSE) {
      if ( (curNest[Nest.ctrl])[Loop.i] == "i") BREAK;
      loopNests = loopNests? (curNest  loopNests) : curNest;
   }
   switch (input)
   {
    case _: input
   }
} />

