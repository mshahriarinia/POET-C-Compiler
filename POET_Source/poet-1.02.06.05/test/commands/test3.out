
DEBUGGING at level 1
a=5
===>5
if a>0===>1
b=a+1
===>6
"abc":EXP
===>0
5:EXP
===>1
END DEBUGGING at level 1

DEBUGGING at level 1
input=(1 2 3 4 7)
===>(1 2 3 4 7)
inputp=input
===>(1 2 3 4 7)
FOR init: inputp=input===>(1 2 3 4 7)
FOR cond: inputp!=""===>1
HEAD(inputp)
===>1
curelem=HEAD(inputp)
===>1
if curelem==1===>1
CONTINUE;
TAIL(inputp)
===>(2 3 4 7)
inputp=TAIL(inputp)
===>(2 3 4 7)
FOR incr: inputp=TAIL(inputp)===>(2 3 4 7)
FOR cond: inputp!=""===>1
HEAD(inputp)
===>2
curelem=HEAD(inputp)
===>2
if curelem==1===>0
if curelem==3===>0
TAIL(inputp)
===>(3 4 7)
inputp=TAIL(inputp)
===>(3 4 7)
FOR incr: inputp=TAIL(inputp)===>(3 4 7)
FOR cond: inputp!=""===>1
HEAD(inputp)
===>3
curelem=HEAD(inputp)
===>3
if curelem==1===>0
if curelem==3===>1
BREAK;
END DEBUGGING at level 1

DEBUGGING at level 1
input=Nest#(Loop#("i",0,100,1),Nest#(Loop#("j",1,10,1),Stmt#1))
===>Nest#(Loop#("i",0,100,1),Nest#(Loop#("j",1,10,1),Stmt#1))
Entering FOREACH(Nest#(Loop#("i",0,100,1),Nest#(Loop#("j",1,10,1),Stmt#1)))
FOREACH found : Loop#("i",0,100,1)
if curLoop[Loop.i]=="j"===>0
FOREACH found : Loop#("j",1,10,1)
if curLoop[Loop.i]=="j"===>1
CONTINUE;
Exiting FOREACH(Nest#(Loop#("i",0,100,1),Nest#(Loop#("j",1,10,1),Stmt#1)))
loopNests=""
===>""
Entering FOREACH(Nest#(Loop#("i",0,100,1),Nest#(Loop#("j",1,10,1),Stmt#1)))
FOREACH found : Nest#(Loop#("i",0,100,1),Nest#(Loop#("j",1,10,1),Stmt#1))
if curNest[Nest.ctrl][Loop.i]=="i"===>1
BREAK;
Exiting FOREACH(Nest#(Loop#("i",0,100,1),Nest#(Loop#("j",1,10,1),Stmt#1)))
CASE cond: input===>Nest#(Loop#("i",0,100,1),Nest#(Loop#("j",1,10,1),Stmt#1))
CASE Nest#(Loop#("i",0,100,1),Nest#(Loop#("j",1,10,1),Stmt#1)) matched label: _===>Nest#(Loop#("i",0,100,1),Nest#(Loop#("j",1,10,1),Stmt#1))
END DEBUGGING at level 1
